tests/nested/core20/tpm/task.yaml

```
@@ -52,3 +52,17 @@ execute: |
 
     # check for sealed keys marker
     nested_exec "test -f /var/lib/snapd/device/fde/sealed-keys"
+
+    # check that kernel command line is properly recorded
+    cmdline_modeenv="$(grep current_kernel_command_lines= < modeenv  | sed -e 's/current_kernel_command_lines=//' | jq -r '.[0]')"
+    cmdline="$(nested_exec "cat /proc/cmdline")"
+    test "$cmdline" = "$cmdline_modeenv"
+    echo "$cmdline" | MATCH "snapd_recovery_mode=run "
+
+    # check that kernel command lines is restored when booted successfully
+    nested_exec "sudo sed -i -e 's/current_kernel_command_lines=.*/current_kernel_command_lines=/' /var/lib/snapd/modeenv"
+    boot_id="$( nested_get_boot_id )"
+    nested_exec "sudo reboot" || true
+    nested_wait_for_reboot "${boot_id}"
+    cmdline_modeenv_restored="$(grep current_kernel_command_lines= < modeenv  | sed -e 's/current_kernel_command_lines=//' | jq -r '.[0]')"
```

this check is wrong, we don't modify the `$PWD/modenv` file here from above, so when we the sed we do it on `/var/lib/snapd/modeenv`, and don't modify `$PWD/modeenv` that is being used here. I think this needs to be doing something like

```suggestion
    cmdline_modeenv_restored="$(grep current_kernel_command_lines= < /var/lib/snapd/modeenv  | sed -e 's/current_kernel_command_lines=//' | jq -r '.[0]')"
```

---

updated